.github/workflows/benchmarks.yml

```diff
@@ -1,83 +1,45 @@
     strategy:
       matrix:
-        os: [ubuntu-20.04]
+        os: [ubuntu-22.04]
 
-    steps:
+    name: Gate benchmarks
+    runs-on: ${{ matrix.os }}
 
-      - name: Checkout PennyLane-Lightning master
-        uses: actions/checkout@v2
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
-          ref: master
-          path: main
+          access_token: ${{ github.token }}
+
+      - name: Checkout PennyLane-Lightning
+        uses: actions/checkout@v3
 
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
         name: Install Python
         with:
           python-version: '3.7'
 
       - name: Install dependencies
-        run: sudo apt-get -y -q install cmake gcc
-
-      - name: Get required Python packages
-        run: |
-          cd main
-          python -m pip install --upgrade pip
-          pip install -r requirements-dev.txt
-          pip install matplotlib
-
-      - name: Install lightning.qubit device (master)
-        run: |
-          cd main
-          python3 setup.py build_ext -i --define="CMAKE_CXX_COMPILER=$(which g++-10)"
-          pip install -e .
-
-      - name: Benchmark lightning master device
-        run: |
-          cd main
-          python .github/workflows/benchmarks/run_bench.py lightning.qubit lightning_master.json
+        run: sudo apt-get update && sudo apt-get -y -q install cmake gcc-10 g++-10 libopenblas-dev
 
-      - name: Benchmark default qubit device
+      - name: Build GBenchmark
         run: |
-          cd main
-          python .github/workflows/benchmarks/run_bench.py default.qubit default_qubit.json
+            cmake pennylane_lightning/src/ -BBuildGBench -DBUILD_BENCHMARKS=ON -DENABLE_BLAS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER="$(which g++-$GCC_VERSION)"
+            cmake --build ./BuildGBench  --parallel 2
 
-      - name: Checkout PennyLane-Lightning PR
-        uses: actions/checkout@v2
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-          path: pr
-
-      - name: Install lightning.qubit device (PR)
+      - name: Run GBenchmark
         run: |
-          cd main
-          make clean || true
-          pip uninstall pennylane-lightning -y
-          cd ../pr
-          python3 setup.py build_ext -i --define="CMAKE_CXX_COMPILER=$(which g++-10)"
-          pip install -e .
-
-      - name: Benchmark lightning PR device
-        run: |
-          cd pr
-          python .github/workflows/benchmarks/run_bench.py lightning.qubit lightning_pr.json
-          mv lightning_pr.json ../main
-
-      - name: Plot results
-        run: |
-          cd main
-          python .github/workflows/benchmarks/plot_results.py
-
-      - uses: actions/upload-artifact@v2
-        with:
-          name: gates.png
-          path: ./main/gates.png
+            ./BuildGBench/benchmarks/utils
+            ./BuildGBench/benchmarks/apply_operations
```

Comment:
As @chaeyeunpark suggested, we can use a regex here to select a subset of the tests to ensure things work.